internal/js/modules/k6/browser/common/page.go

```diff
@@ -227,17 +226,20 @@ type responseWaiter struct {
 
 func NewResponseEventHandler() *ResponseEventHandler {
 	return &ResponseEventHandler{
-		activeWaiters: make(map[string]*responseWaiter),
+		activeWaiters: make(map[int64]*responseWaiter),
 	}
 }
 
-func (reh *ResponseEventHandler) addWaiter(waiter *responseWaiter) {
+func (reh *ResponseEventHandler) addWaiter(waiter *responseWaiter) int64 {
 	reh.mu.Lock()
 	defer reh.mu.Unlock()
-	reh.activeWaiters[waiter.id] = waiter
+
+	id := reh.generateWaiterID()
```

Comment:
Can we move `id := reh.generateWaiterID()` to before `reh.mu.Lock()`? Avoids a situation where this is holding onto two locks.